src/components/typography/text/text.js

```diff
@@ -33,14 +36,58 @@ const Headline = props => {
 
 Headline.displayName = 'TextHeadline';
 Headline.propTypes = {
-  elementType: PropTypes.oneOf(['h1', 'h2', 'h3']).isRequired,
+  as(props, propName, componentName, ...rest) {
+    // is required if the prop 'elementType' is not defined
+    if (!props.elementType) {
+      return PropTypes.oneOf(['h1', 'h2', 'h3']).isRequired(
+        props,
+        propName,
+        componentName,
+        ...rest
+      );
+    }
+    return PropTypes.oneOf(['h1', 'h2', 'h3'])(
+      props,
+      propName,
+      componentName,
+      ...rest
+    );
+  },
+  elementType(props, propName, componentName, ...rest) {
+    if (props[propName] != null) {
+      const message = oneLine`
+            "${propName}" property of "${componentName}" has been deprecated
+            and will be removed in the next major version.\n
+            Please use "as" prop instead.`;
+      warning(false, message);
```

Comment:
Starting a new thread on using `warning` here. We can just return an `Error` to my knowledge.

Example from the official React docs:

```js
customProp: function(props, propName, componentName) {
    if (!/matchme/.test(props[propName])) {
      return new Error(
        'Invalid prop `' + propName + '` supplied to' +
        ' `' + componentName + '`. Validation failed.'
      );
    }
  },
```

Reply:
I'd prefer to use a Warning rather than an Error here because it's **not invalid** to use the prop, but the developer should still be _warned_ about the future deprecation to be encouraged to migrate to the new prop.

I also ~stole it~ took inspiration from the official React docs:
https://reactjs.org/warnings/dont-call-proptypes.html#fixing-the-false-positive-in-third-party-proptypes

Reply:
Ah cool. Thanks. Will resolve then.